.github/workflows/docs.yml

```diff
@@ -20,19 +20,45 @@ on:
     - cron: "17 17 * * 3,6"
 
 jobs:
+  cache-refresh:
+    name: Refresh Cache ♽
+    runs-on: ubuntu-latest
+    if: github.event_name == 'schedule'
+    steps:
+      - name: Refresh Cache ♽
+        run: |
+          gh extension install actions/gh-actions-cache
+          echo "Fetching list of cache keys..."
+          cacheKeysForPR=$(gh actions-cache list -R $REPO -B $BRANCH -L 100 | cut -f 1 )
+          ## Setting this to not fail the workflow while deleting cache keys.
+          set +e
+          echo "Deleting caches..."
+          for cacheKey in $cacheKeysForPR
+          do {
+            gh actions-cache delete $cacheKey -R $REPO -B $BRANCH --confirm
+          }
+          done
+          echo "Cache refresh completed."
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REPO: ${{ github.repository }}
+          BRANCH: ${{ github.event.repository.default_branch }}
+
   publish-dev:
     name: Build & Deploy TLGC (dev) 📖
     runs-on: ubuntu-latest
     container:
       image: ghcr.io/insightsengineering/rstudio:latest
+    needs: cache-refresh
```

Comment:
I hope this will execute properly when the `cache-refresh` job is not in the workflow, when the workflow runs *not* on schedule (or maybe it will be there but skipped?)